source/fuzz/fuzzer_pass_donate_modules.cpp

```diff
@@ -308,18 +325,33 @@ void FuzzerPassDonateModules::HandleTypesAndValues(
       case SpvOpTypeStruct: {
         // Similar to SpvOpTypeArray.
         std::vector<uint32_t> member_type_ids;
+        bool can_donate = true;
         for (uint32_t i = 0; i < type_or_value.NumInOperands(); i++) {
           auto component_type_id = type_or_value.GetSingleWordInOperand(i);
+          if (!original_id_to_donated_id->count(component_type_id)) {
+            // We did not donate every member type for this struct type, so we
+            // cannot donate the struct type.
+            can_donate = false;
+            break;
```

Comment:
I wonder if it would be better to add another function and change "continue"->"return" (elsewhere). Then this could also be a "return".

Reply:
Good idea.

Reply:
Done (and it is now much simpler!)